src/plugin/runtime-support.cc

```diff
@@ -427,49 +429,86 @@ void rt::protocol_begin(const std::optional<std::string>& analyzer) {
 
         if ( ! c->analyzer->AddChildAnalyzer(child) )
             // Same comment as above re/ ignoring the error and memory mgmt.
-            return;
+            return child->GetID();
 
         child->FirstPacket(true, nullptr);
         child->FirstPacket(false, nullptr);
+
+        return {};
     }
 }
 
-void rt::protocol_data_in(const hilti::rt::Bool& is_orig, const hilti::rt::Bytes& data) {
+void rt::protocol_data_in(const hilti::rt::Bool& is_orig, const hilti::rt::Bytes& data,
+                          const std::optional<rt::ProtocolHandle>& h) {
     auto cookie = static_cast<Cookie*>(hilti::rt::context::cookie());
     assert(cookie);
 
     auto c = std::get_if<cookie::ProtocolAnalyzer>(cookie);
     if ( ! c )
         throw ValueUnavailable("no current connection available");
 
-    c->analyzer->ForwardStream(data.size(), reinterpret_cast<const u_char*>(data.data()), is_orig);
+    auto len = data.size();
+    auto* data_ = reinterpret_cast<const u_char*>(data.data());
+
+    if ( h ) {
+        auto* child = c->analyzer->FindChild(h->id);
+        if ( ! child )
+            throw ValueUnavailable(hilti::rt::fmt("unknown child analyzer %s", *h));
+
+        child->DeliverStream(len, data_, is_orig);
```

Comment:
Since we have access to the output handler and can trigger the lifecycle management of the child list, it seems we can implement this from the outside.